extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProvider.java

```diff
@@ -213,14 +189,28 @@ private void fetchNextBatch() throws IOException {
                             if (path.contains(REPEATED_URI_SEPARATOR)) {
                                 path = REPEATED_URI_SEPARATOR_PATTERN.matcher(path).replaceAll(URI_SEPARATOR);
                             }
+                            final URI uri;
                             try {
-                                return new URI(S3_URI_SCHEME, directory.getUserInfo(), directory.getHost(),
+                                uri = new URI(S3_URI_SCHEME, directory.getUserInfo(), directory.getHost(),
                                         directory.getPort(), path, null, null);
                             } catch (final URISyntaxException e) {
                                 throw new UncheckedDeephavenException("Failed to create URI for S3 object with key: "
                                         + s3Object.key() + " and bucket " + bucketName + " inside directory "
                                         + directory, e);
                             }
+                            uriToFileSize.compute(uri, (key, existingSize) -> {
+                                if (existingSize == null) {
+                                    // TODO (@Ryan) This map will keep growing as more files are discovered with this
```

Comment:
It's not "really" OK. But on the other hand, maybe it's not that big of a deal, either? I love the idea of saving an rpc per file. We could hold the map in a `SoftReference` if we get worried.